diff -rup elogind-252.9/src/login/logind-dbus.c elogind-252.9-silence/src/login/logind-dbus.c
--- elogind-252.9/src/login/logind-dbus.c	2023-05-15 13:35:29.000000000 +0200
+++ elogind-252.9-silence/src/login/logind-dbus.c	2023-06-10 17:41:00.374974384 +0200
@@ -864,7 +864,6 @@ static int method_create_session(sd_bus_
                 /* Wut? There's already a session by this name and we didn't find it above? Weird, then let's
                  * not trust the audit data and let's better register a new ID */
                 if (hashmap_contains(m->sessions, id)) {
-                        log_warning("Existing logind session ID %s used by new audit session, ignoring.", id);
                         audit_id = AUDIT_SESSION_INVALID;
                         id = mfree(id);
                 }
diff -rup elogind-252.9/src/login/logind-session.c elogind-252.9-silence/src/login/logind-session.c
--- elogind-252.9/src/login/logind-session.c	2023-05-15 13:35:29.000000000 +0200
+++ elogind-252.9-silence/src/login/logind-session.c	2023-06-10 17:40:26.705058445 +0200
@@ -834,13 +834,6 @@ int session_start(Session *s, sd_bus_mes
         if (r < 0)
                 return r;
 
-        log_struct(s->class == SESSION_BACKGROUND ? LOG_DEBUG : LOG_INFO,
-                   "MESSAGE_ID=" SD_MESSAGE_SESSION_START_STR,
-                   "SESSION_ID=%s", s->id,
-                   "USER_ID=%s", s->user->user_record->user_name,
-                   "LEADER="PID_FMT, s->leader,
-                   LOG_MESSAGE("New session %s of user %s.", s->id, s->user->user_record->user_name));
-
         if (!dual_timestamp_is_set(&s->timestamp))
                 dual_timestamp_get(&s->timestamp);
 
@@ -986,18 +979,6 @@ int session_finalize(Session *s) {
         if (!s->user)
                 return -ESTALE;
 
-        if (s->started)
-                log_struct(s->class == SESSION_BACKGROUND ? LOG_DEBUG : LOG_INFO,
-                           "MESSAGE_ID=" SD_MESSAGE_SESSION_STOP_STR,
-                           "SESSION_ID=%s", s->id,
-                           "USER_ID=%s", s->user->user_record->user_name,
-                           "LEADER="PID_FMT, s->leader,
-                           LOG_MESSAGE("Removed session %s.", s->id));
-#if 1 /// let elogind at least put out a debug message if s was not started
-        else
-                log_debug_elogind("Session %s not started, finalizing...", s->id);
-#endif // 1
-
         s->timer_event_source = sd_event_source_unref(s->timer_event_source);
 
         if (s->seat)
