
Silence annoying session logging.

diff -rup elogind-239.1/src/login/logind-dbus.c elogind-239.1-silent/src/login/logind-dbus.c
--- elogind-239.1/src/login/logind-dbus.c	2018-10-17 19:50:46.000000000 +0200
+++ elogind-239.1-silent/src/login/logind-dbus.c	2018-11-16 19:19:40.890552583 +0100
@@ -759,7 +759,7 @@ static int method_create_session(sd_bus_
                  * the audit data and let's better register a new
                  * ID */
                 if (hashmap_get(m->sessions, id)) {
-                        log_warning("Existing logind session ID %s used by new audit session, ignoring.", id);
+                        /* log_warning("Existing logind session ID %s used by new audit session, ignoring.", id); */
                         audit_id = AUDIT_SESSION_INVALID;
                         id = mfree(id);
                 }
diff -rup elogind-239.1/src/login/logind-session.c elogind-239.1-silent/src/login/logind-session.c
--- elogind-239.1/src/login/logind-session.c	2018-10-17 19:50:46.000000000 +0200
+++ elogind-239.1-silent/src/login/logind-session.c	2018-11-16 19:20:18.090863797 +0100
@@ -639,13 +639,6 @@ int session_start(Session *s, sd_bus_mes
         if (r < 0)
                 return r;
 
-        log_struct(s->class == SESSION_BACKGROUND ? LOG_DEBUG : LOG_INFO,
-                   "MESSAGE_ID=" SD_MESSAGE_SESSION_START_STR,
-                   "SESSION_ID=%s", s->id,
-                   "USER_ID=%s", s->user->name,
-                   "LEADER="PID_FMT, s->leader,
-                   LOG_MESSAGE("New session %s of user %s.", s->id, s->user->name));
-
         if (!dual_timestamp_is_set(&s->timestamp))
                 dual_timestamp_get(&s->timestamp);
 
@@ -778,14 +771,6 @@ int session_finalize(Session *s) {
         if (!s->user)
                 return -ESTALE;
 
-        if (s->started)
-                log_struct(s->class == SESSION_BACKGROUND ? LOG_DEBUG : LOG_INFO,
-                           "MESSAGE_ID=" SD_MESSAGE_SESSION_STOP_STR,
-                           "SESSION_ID=%s", s->id,
-                           "USER_ID=%s", s->user->name,
-                           "LEADER="PID_FMT, s->leader,
-                           LOG_MESSAGE("Removed session %s.", s->id));
-
         s->timer_event_source = sd_event_source_unref(s->timer_event_source);
 
         if (s->seat)
