diff -rup openssh-8.5p1/sshconnect2.c openssh-8.5p1-silence/sshconnect2.c
--- openssh-8.5p1/sshconnect2.c	2021-03-04 09:29:44.850959279 +0100
+++ openssh-8.5p1-silence/sshconnect2.c	2021-03-04 09:31:43.453391045 +0100
@@ -512,7 +512,6 @@ ssh_userauth2(struct ssh *ssh, const cha
 			memcpy(&myproposal, &myproposal_default, sizeof(myproposal));
 			myproposal[PROPOSAL_ENC_ALGS_STOC] = "none";
 			myproposal[PROPOSAL_ENC_ALGS_CTOS] = "none";
-			fprintf(stderr, "WARNING: ENABLED NONE CIPHER!!!\n");
 			/* NONEMAC can only be used in context of the NONE CIPHER */
 			if (options.nonemac_enabled == 1) {
 				myproposal[PROPOSAL_MAC_ALGS_STOC] = "none";
