From 82707511635de59d29db53ac1916414ae390b032 Mon Sep 17 00:00:00 2001
From: arvidn <arvid@libtorrent.org>
Date: Sun, 20 Jan 2019 17:59:11 +0100
Subject: [PATCH] on linux, link against lbdl when using openssl

---
 Jamfile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Jamfile b/Jamfile
index a64ffe0193..2928add3db 100644
--- a/Jamfile
+++ b/Jamfile
@@ -806,6 +806,7 @@ lib torrent
 	<link>shared:<define>BOOST_SYSTEM_SOURCE
 	<crypto>openssl:<library>ssl
 	<crypto>openssl:<library>crypto
+	<crypto>openssl,<target-os>linux:<library>dl
 
 	<dht>on:<source>src/kademlia/$(KADEMLIA_SOURCES).cpp
 	<dht>on:<source>ed25519/src/$(ED25519_SOURCES).cpp
