From 5a463bc4714ada941e8627a9aedec1d551e870d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Holger=20Hoffst=C3=A4tte?= <holger@applied-asynchrony.com>
Date: Thu, 18 Jun 2020 23:18:16 +0200
Subject: [PATCH] Add timestamps to logfile output when not using syslog.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Holger Hoffstätte <holger@applied-asynchrony.com>
---
 report.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/report.c b/report.c
index f93b92eb..f4c5b3d0 100644
--- a/report.c
+++ b/report.c
@@ -84,8 +84,16 @@ void report (FILE *errfp, const char *message, ...)
 	    fputc( '\n', errfp );
 	    ++message;
 	}
-	if (!partial_suppress_tag)
-		fprintf (errfp, "%s: ", program_name);
+	if (!partial_suppress_tag) {
+		time_t timer;
+		char buffer[26];
+		struct tm* tm_info;
+		timer = time(NULL);
+		tm_info = localtime(&timer);
+		mktime(tm_info);
+		strftime (buffer, sizeof(buffer), "%b %d %H:%M:%S", tm_info);
+		fprintf (errfp, "%s %s: ", buffer, program_name);
+	}
 	partial_suppress_tag = 0;
 
 	va_start (args, message);
-- 
GitLab

