diff -rup zsh-5.7.1/Completion/Unix/Type/_tar_archive zsh-5.7.1-zstd/Completion/Unix/Type/_tar_archive
--- zsh-5.7.1/Completion/Unix/Type/_tar_archive	2018-06-24 02:14:58.000000000 +0200
+++ zsh-5.7.1-zstd/Completion/Unix/Type/_tar_archive	2019-03-08 18:24:36.940735850 +0100
@@ -21,7 +21,7 @@ if [[ "$1" = *[urtx]* ]]; then
   elif [[ "$1" = *J* ]]; then
     _files "$expl[@]" -g '*.(tar|TAR).(lzma|xz)(-.)'
   elif [[ "$_cmd_variant[$service]" == (gnu|libarchive) ]]; then
-    _files "$expl[@]" -g '*.((tar|TAR)(.gz|.GZ|.Z|.bz2|.lzma|.xz|)|(tbz|tgz|txz))(-.)'
+    _files "$expl[@]" -g '*.((tar|TAR)(.gz|.GZ|.Z|.bz2|.lzma|.xz|.zst)|(tbz|tgz|txz))(-.)'
   else
     _files "$expl[@]" -g '*.(tar|TAR)(-.)'
   fi
