--- sonnet-5.89.0/src/core/loader.cpp	2021-12-04 19:55:29.000000000 +0100
+++ sonnet-5.89.0-nowarn/src/core/loader.cpp	2021-12-13 01:33:35.532745595 +0100
@@ -83,10 +83,8 @@ SpellerPlugin *Loader::createSpeller(con
     auto clientsItr = d->languageClients.constFind(plang);
     if (clientsItr == d->languageClients.constEnd()) {
         if (language.isEmpty()) {
-            qCWarning(SONNET_LOG_CORE) << "No language dictionaries for the language:" << plang << "trying to load en_US as default";
             return createSpeller(QStringLiteral("en_US"), clientName);
         }
-        qCWarning(SONNET_LOG_CORE) << "No language dictionaries for the language:" << plang;
         Q_EMIT loadingDictionaryFailed(plang);
         return nullptr;
     }
@@ -102,7 +100,6 @@ SpellerPlugin *Loader::createSpeller(con
                 return client->name() == backend;
             });
             if (unknown) {
-                qCWarning(SONNET_LOG_CORE) << "Default client" << backend << "doesn't support language:" << plang;
                 backend = QString();
             }
         }
@@ -126,7 +123,6 @@ SpellerPlugin *Loader::createSpeller(con
         }
     }
 
-    qCWarning(SONNET_LOG_CORE) << "The default client" << backend << "has no language dictionaries for the language:" << plang;
     return nullptr;
 }
 
@@ -281,10 +277,6 @@ void Loader::loadPlugins()
             loadPlugin(dir.absoluteFilePath(fileName));
         }
     }
-
-    if (d->loadedPlugins.isEmpty()) {
-        qCWarning(SONNET_LOG_CORE) << "Sonnet: No speller backends available!";
-    }
 #else
 #ifdef Q_OS_MACOS
     loadPlugin(QString());
