diff -rup rsync-3.1.2.orig/rsync.h rsync-3.1.2/rsync.h
--- rsync-3.1.2.orig/rsync.h	2015-08-08 21:47:03.000000000 +0200
+++ rsync-3.1.2/rsync.h	2017-11-15 00:17:05.719235817 +0100
@@ -135,7 +135,7 @@
 #define WRITE_SIZE (32*1024)
 #define CHUNK_SIZE (32*1024)
 #define MAX_MAP_SIZE (256*1024)
-#define IO_BUFFER_SIZE (32*1024)
+#define IO_BUFFER_SIZE (256*1024)
 #define MAX_BLOCK_SIZE ((int32)1 << 17)
 
 /* For compatibility with older rsyncs */
