diff -rup rsync-3.1.2.orig/receiver.c rsync-3.1.2/receiver.c
--- rsync-3.1.2.orig/receiver.c	2015-09-07 19:07:17.000000000 +0200
+++ rsync-3.1.2/receiver.c	2017-11-15 00:17:09.555937716 +0100
@@ -245,7 +245,7 @@ static int receive_data(int f_in, char *
 	OFF_T preallocated_len = 0;
 #endif
 
-	if (preallocate_files && fd != -1 && total_size > 0 && (!inplace || total_size > size_r)) {
+	if (preallocate_files && fd != -1 && total_size >= (128*1024*1024) && total_size > size_r) {
 		/* Try to preallocate enough space for file's eventual length.  Can
 		 * reduce fragmentation on filesystems like ext4, xfs, and NTFS. */
 		if (do_fallocate(fd, 0, total_size) == 0) {
