diff -rup systemd-242/src/udev/net/link-config.c systemd-242-nowarn/src/udev/net/link-config.c
--- systemd-242/src/udev/net/link-config.c	2019-04-11 18:28:36.000000000 +0200
+++ systemd-242-nowarn/src/udev/net/link-config.c	2019-04-15 14:46:50.636710744 +0200
@@ -201,7 +201,6 @@ int link_config_load(link_config_ctx *ct
 
         if (!enable_name_policy()) {
                 ctx->enable_name_policy = false;
-                log_info("Network interface NamePolicy= disabled on kernel command line, ignoring.");
         }
 
         /* update timestamp */
